### ISSUE TITLE
WEP (arp replay attack) seems to stop sending replay packets after ~ 60 sec

### ISSUE THREAD

**Issue by vom513** [derv82/wifite2#21]
This is quite the hair puller.  It seems that after approx. 60 seconds, replay packets stop flowing.  I'm testing this against an older Cisco 1230 series AP (it runs IOS, so I can look at all the gory client details).

What I see is that the attack starts, and once the first arp packet is heard, replay begins (so far - textbook).  I then monitor the client stats on the AP paying attention to the last activity counter.  This constantly reads 0 seconds (ago) as the replay packets are streaming from the laptop.  After about 60 sec or so - this counter starts incrementing - meaning that the "client" is idle.  There is no disassocation on the AP or anything - wifite just shows that IVs aren't incrementing at the previous rate (essentially a trickle at this point).  Looking at ps - aireplay is running fine.  I've also compared the command line args. of this aireplay to what it looks like in original wifite - and it looks like the same command line.

I saw you also updated "original" wifite.  The same 60 sec behavior is observed there.

I've tested this on the "original" wifite (r87 ?) - and it works as expected.  In fact - I even lowered the pps to 200 to force it to run longer.  At that rate, original wifite ran for ~ 4 min. to collect ~ 31k IVs and crack the (128 bit) WEP.

Is there anything in wifite2 that happens at around the 60 sec mark that might do something to the interface causing replays to stop ?  As I said - ps shows it still running, so I'm not sure what could happen to that running process to make it stop sending.  Since original wifite and running the aircrack suite by hand both work as expected consistently - it seems like something in wifite2 is causing the replay to be interrupted.

Anything I could do to further debug and troubleshoot this ?

PS: Thanks for continuing to work on this project.

**Comment by derv82** [comment]
I can reproduce this easily by running the `arpreplay` attack at a very-low PacketsPerSecond:

`./Wifite.py --wep --pps 10`

> IVs aren't incrementing at the previous rate (essentially a trickle at this point).

The CPU usage of the `aireplay-ng` process decreases gradually, then drops when the packets stop.

I'll look into this...

**Comment by derv82** [comment]
Ok, I don't see anything that would cause the command to stop after 60 seconds... And it's not *exactly* 60 seconds for me (more like 2minutes).

Looking at [the `subprocess.Popen` documentation](https://docs.python.org/2/library/subprocess.html#subprocess.Popen.wait):

> **Warning**
> This will deadlock when using `stdout=PIPE` and/or `stderr=PIPE` and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data.

The script *does* use `stdout=PIPE`, and aireplay-ng *does* generate a lot of output when replaying packets...

I made a change to redirect stdout to `/dev/null` for the aireplay attacks. It seems to work for `arpreplay` -- I got to 10,000 IVS at 10 PPS (took ~20 minutes).

But this change may break other aireplay-ng attacks.... There's no option to specify the `aireplay-ng` attacks (`--chopchop`, `--fragment`, etc), so it's hard for me to test. I'll leave this open until the issue (arpreplay PPS degradation) is resolved.
